pkg/scale/encode.go

```diff
@@ -327,7 +450,11 @@ func (es *encodeState) encodeStruct(in interface{}) (err error) {
 
 // encodeLength is a helper function that calls encodeUint, which is the scale length encoding
 func (es *encodeState) encodeLength(l int) (err error) {
-	return es.encodeUint(uint(l))
+	err = es.encodeUint(uint(l))
+	if err != nil {
+		err = fmt.Errorf("encoding Length: %w", err)
```

Comment:
the context should not be this, the caller should add this `encoding length: %w` context instead. Here it should just be

```suggestion
		err = fmt.Errorf("encoding uint: %w", err)
```